app/src/main/java/woowacourse/paint/model/drawable/shape/DrawableSquare.kt

```diff
@@ -0,0 +1,38 @@
+package woowacourse.paint.model.drawable.shape
+
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.RectF
+import woowacourse.paint.model.drawable.DrawableElement
+
+data class DrawableSquare(
+    private val rect: RectF = RectF(),
+    override val paint: Paint,
+) : DrawableShape {
+
+    init {
+        paint.apply { style = Paint.Style.FILL }
+    }
+
+    override fun drawCurrent(canvas: Canvas) {
+        canvas.drawRect(rect, paint)
+    }
+
+    override fun startDrawing(x: Float, y: Float): DrawableElement {
+        return DrawableSquare(
+            RectF(x, y, x, y),
+            paint = Paint(paint),
+        )
+    }
+
+    override fun keepDrawing(x: Float, y: Float) {
+        rect.right = x
+        rect.bottom = y
+    }
+
+    override fun changePaintColor(color: Int): DrawableElement {
+        return copy(
+            paint = Paint(paint).apply { this.color = color },
```

Comment:
이 부분도 scope 함수가 꼭 필요할까 생각이 들어요! 하티의 의견이 궁금합니당

Reply:
scope 함수가 꼭 필요한 건 아니지만, 코드를 간결하게 작성하고 싶어서 scope 함수를 사용했어요 !

```suggestion
        val newPaint = Paint(paint)
        newPaint.color = color
        return copy(paint = newPaint)
```
scope 함수 없이 사용한다면 이렇게 코드가 작성될 것 같아요.
다만, 저는 위 코드보다는 기존 코드 방식을 선호하는데요! 핑구의 생각은 어떠신가요??

Reply:
아하 그렇네요! 위와 같은 경우인 줄 알고 자세히 안보고 바로 달아버렸어요 죄송해요😂